doc/source/tool.rst

```diff
@@ -0,0 +1,643 @@
+customization you will want to do (such as changing the number of spaces
+to indent) can be done with the printing options. However, if you really
+need more control over the output, you can make changes to the ``dpp``
+entries in the file `dpp.lisp <https://github.com/dylan-lang/lisp-to-dylan/tree/master/code/dpp.lisp>`__. Here are four
+example entries to give you an idea of how ``dpp`` works:
+
+::
+
+   (dpp 'symbol   dpp-symbol)
+   (dpp (if)      dpp-conditional)
+   (dpp (:else)   dpp-unindented)
+   (dpp (begin)   (format s "~@<begin~W~:>" `(:body (end) ,@(rest x))))
+
+The first example says that all expressions of type ``symbol`` will be
+handled by calling the function ``dpp-symbol`` with two arguments: the
+stream to print on and the symbol to print. In place of ``'symbol``, you
+can use any (quoted) Common Lisp type specifier.
+
+In the second example, ``(if)`` is used to indicate a list whose first
+element is ``if``. The example says that all such lists are printed by a
+function named ``dpp-conditional``. Again, the arguments are the stream
+and the expression.
+
+The third example is very similar; its purpose is to show the convention
+that elements of the Dylan parse tree that represent parts of statements
+start with keyword symbols (such as ``:else``), whereas those that
+represent complete statements (such as ``if``) are non-keyword symbols.
+
+The fourth example says that all lists that start with the symbol
+``begin`` will be printed by calling the function
+
+::
+
+   (lambda (s x) (format s "~@<begin~W~:>" `(:body (end) ,@(rest x))))
+
+That is, the expression is called with ``s`` bound to the stream and
+``x`` bound to the expression to print. There are two things worth
+discussing in this example.
+
+First, the expression ``(:body (end) ,@(rest x))`` is the
+internal parse tree representation for a body that should be followed by
+just an "end" (rather than, say, "end method" or "end method name").
+
+Second, the format string ``"~@<begin~W~:>"`` makes use of
+pretty-printing format directives from `Chapter
+27 <https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node253.html#SECTION003100000000000000000>`__
+of *Common Lisp the Language: 2nd Edition*. You will need to understand
+this chapter before you can make changes to DPP.
+
+Please observe the convention that printing blocks surround a whole
+statement; the ``dpp-body`` function does *not* start a new printing
+block. Sub-parts may be put in blocks.
+
+To Do List
+----------
+
+LTD has been tested on about 35,000 lines (1.1MB in 39 files) of Lisp
+code (in the `test <https://github.com/dylan-lang/lisp-to-dylan/tree/master/test/>`__ directory) taken from an archive of
+`Software from AI
+books <https://web.archive.org/web/20030803095450/http://yoda.cis.temple.edu:8080/UGAIWWW/books/index.html>`__ and
+several other sources. By "tested", I mean that LTD runs without
+crashing on two platforms (Lispworks and Lucid Common Lisp, on a Sparc
```

Comment:
If we are treating this as a living document we might want to add a bit of context to the historical parts